src/azure-cli/azure/cli/command_modules/rdbms/_client_factory.py

```diff
@@ -38,7 +38,7 @@ def get_mariadb_management_client(cli_ctx, **_):
         return MariaDBManagementClient(
             subscription_id=getenv(SUB_ID_OVERRIDE),
             base_url=rm_uri_override,
-            credentials=credentials)
+            credential=credentials)
```

Comment:
There're huge changes from track1 credentials to track2 credential. I'm afraid the origin credential constructor can't work with track2. Have you tested?

Reply:
Sorry, I totally missed this point. 

Will this migration include azure-identity library? 

Looks like now the python sdk suggests using ClientSecretCredential instead of ServicePrincipalCredentials (https://docs.microsoft.com/en-us/azure/developer/python/azure-sdk-authenticate?tabs=cmd) . Testing with ServicePrincipalCredentials didn't work but ClientSecretCredential worked well.. 

Let me know if I am allowed to replace ServicePrincipalCredentials with ClientSecretCredential..

Reply:
@jiasli is expert of cli auth. Maybe he can answer this question😃

Reply:
Yes, when running with the ServicePrincipalCredentials, the error I was seeing was 
AttributeError: 'ServicePrincipalCredentials' object has no attribute 'get_token'..!

Reply:
Thank you so much for detailed explanation! Made the change per your comment

Reply:
Sorry to bother @jiasli but the style error is telling me 
src/azure-cli/azure/cli/command_modules/rdbms/_client_factory.py:29:12: E0611: No name 'identity' in module 'azure' (no-name-in-module)
https://dev.azure.com/azure-sdk/public/_build/results?buildId=776163&view=logs&j=36dd4138-4d53-5e46-00d9-e5cd9744be05&t=1cf3879c-0a42-5ccd-7693-7a4781d739d8&l=134

This import works and style test passes on my local,, so wanted to ask you if there's a known issue/ way to unblock this,,,!!!

Reply:
This is because `dev` branch of Azure CLI doesn't depend on Azure Identity, so it is not installed. Do you have to use Azure Identity for this feature? If so, I can add this dependency.

Reply:
@jiasli Thank you so much!

Reply:
RDBMS module uses get_mgmt_service_client but I think the predecessor also implemented a way to override the authentification using credential. I wanted to keep both ways in case users were using this approach.. thanks!!